### PR TITLE
challenge

### PR DESCRIPTION
**Pull request by wenceslao01**

### CODE REVIEW

File: src/actions/CocktailsActions.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { CHANGE_APP_STATE, GET_COCKTAILS, GET_COCKTAIL_DETAILS, FILTER_COCKTAILS } from './types';
+
+const GET_COCKTAILS_URL = 'https://www.thecocktaildb.com/api/json/v1/1/filter.php?g=Cocktail_glass'
+const GET_COCKTAIL_DETAILS_URL = 'https://www.thecocktaildb.com/api/json/v1/1/lookup.php?i='
+
+export const getCocktails = () => {
+  return (dispatch) => {
+    dispatch({ type: CHANGE_APP_STATE, payload: true });
+    axios.get(GET_COCKTAILS_URL).then(response => {
+      const cocktails = response.data.drinks;
```
Comment (coreyakaplan) [inline src/actions/CocktailsActions.js]: How would you handle a successful response (no errors) with an unexpected payload? IE status code 401.

File: src/components/cocktail-details/CocktailDetails.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+};
+
+const styles = {
+  titleStyle: {
+    fontSize: 16,
+    fontWeight: '600',
+    color: '#000'
+  },
+  imageStyles: {
+    height: 300,
```
Comment (coreyakaplan) [inline src/components/cocktail-details/CocktailDetails.js]: Hard coded dimensions will not display properly on different device sizes. Please describe how you would handle styling for this app on both an ipad, and multiple sized iphones.

File: src/actions/CocktailsActions.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      dispatch({ type: GET_COCKTAILS, payload: cocktails });
+      dispatch({ type: CHANGE_APP_STATE, payload: false });
+    }).catch((err) => {
+      dispatch({ type: CHANGE_APP_STATE, payload: false });
+      console.log('error getting cocktails list from server.', err)
```
[the remark this answers is not in the document]
Comment (wenceslao01) [reply src/actions/CocktailsActions.js]: Depending on the amount of information the user needs and the business logic, i would use a modal, an alert or a view.

for example, if the user is trying to buy something and a purchase fails, i would show a view displaying the error, the purchase status and actions to take (all of this depends on the business logic, of course)

In this case (a list of not sensitive information) i would show a generic server error on the view and, if needed, a button to reload the list.

File: src/actions/CocktailsActions.js (hunk 1 above)
Comment (wenceslao01) [reply src/actions/CocktailsActions.js]: i dont understand the question, if the status code is 401, then the response is not successful, the catch would be executed.

if we need to handle different status codes, then i would use a switch case in the catch callback.

if this question points out that response.data.drinks might not exist, i could use a try catch, and display a proper error in the view.

File: src/components/cocktail-details/CocktailDetails.js (hunk 2 above)
Comment (wenceslao01) [reply src/components/cocktail-details/CocktailDetails.js]: depending on the needed desing, i would use media queries, flex and % units.

**Pull request closed without merging**